cluster-controller/src/main/java/io/strimzi/controller/cluster/resources/TopicControllerConfig.java

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright 2017-2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.controller.cluster.resources;
+
+import io.vertx.core.json.JsonObject;
+
+/**
+ * Represents the configuration for the Topic Controller to deploy
+ */
+public class TopicControllerConfig {
+
+    public static final String NAMESPACE_FIELD = "namespace";
+    public static final String IMAGE_FIELD = "image";
+    public static final String RECONCILIATION_INTERVAL_FIELD = "reconciliationInterval";
+    public static final String ZOOKEEPER_SESSION_TIMEOUT_FIELD = "zookeeperSessionTimeout";
+
+    // Kafka bootstrap servers and Zookeeper nodes can't be specified in the JSON
+    private String kafkaBootstrapServers;
+    private String zookeeperConnect;
+
+    private String namespace;
+    private String image;
```

Comment:
Why is `image` here? The others correspond to the env vars consumed by the tc, but not image.

Fwiw, I don't really understand why this class is distinct from `TopicController` (the subclass of `AbstractCluster`). If it's just representing the env for the tc, you could just use a `Map<String, String>`.

Reply:
I'd like to keep the `topic-controller-config` data field as a separate class for the configuration as it happens for `Storage` class. It needs fields validation as well.
Regarding `image` I agree with you that it's weird having it here but it was for consistency to have the whole JSON representation in the corresponding config class.